Documentation/gettingstarted/k8s-install-openshift-okd.rst

```diff
@@ -0,0 +1,202 @@
+.. only:: not (epub or latex or html)
+
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    https://docs.cilium.io
+
+.. _k8s_install_openshift_okd:
+
+*****************************
+Installation on OpenShift OKD
+*****************************
+
+OpenShift Requirements
+======================
+
+1. Choose preferred cloud provider (this guide was tested in AWS, Azure & GCP)
+
+2. Read `OpenShift documentation <https://docs.okd.io/latest/welcome/index.html>`_ to find out about provider-specific prerequisites
+
+3. `Get OpenShift Installer <https://github.com/openshift/okd#getting-started>`_
+
+.. note:: It's highly recommended to read the docs, unless you have installed
+   OpenShift in the past. Here are a few notes that you may find useful.
+   
+   - with the AWS provider ``openshift-install`` will not work properly
+     when MFA credentials are stored in ``~/.aws/credentials``, traditional credentials are required
+   - with the Azure provider ``openshift-install`` will prompt for
+     credentials and store them in ``~/.azure/osServicePrincipal.json``, it
+     doesn't simply pickup ``az login`` credentials. It's recommended to
+     setup a dedicated service principal and use it
+   - with the GCP provider ``openshift-install`` will only work with a service
+     account key, which has to be set using ``GOOGLE_APPLICATION_CREDENTIALS``
+     environment variable (e.g. ``GOOGLE_APPLICATION_CREDENTIALS=service-account.json``)
+
+Create an OpenShift OKD Cluster
+===============================
+
+First, set cluster name:
+
+.. code:: bash
+
+   CLUSTER_NAME="cluster-1"
+
+Now, create configuration files:
+
+   NOTE: the sample output below is showing the AWS provider, but
+   it should work the same way with other providers
+
+.. code:: bash
+
+   $ openshift-install create install-config --dir "${CLUSTER_NAME}"
+   ? SSH Public Key ~/.ssh/id_rsa.pub
+   ? Platform aws
+   INFO Credentials loaded from default AWS environment variables 
+   ? Region eu-west-1
+   ? Base Domain openshift-test-1.cilium.rocks
+   ? Cluster Name cluster-1
+   ? Pull Secret [? for help] ******************************************************************************************************************************************************************************************
+
+And set ``networkType: Cilium``:
+
+.. code:: bash
+
+   sed -i 's/networkType:\ OVNKubernetes/networkType:\ Cilium/' "${CLUSTER_NAME}/install-config.yaml"
+
+Next, generate OpenShift manifests:
+
+.. code:: bash
+
+   openshift-install create manifests --dir "${CLUSTER_NAME}"
+
+Now, define ``cilium`` namespace:
```

Comment:
I know in the past we have used `cilium` namespace for GKE, but in general would prefer `kube-system` because IIRC scheduling works a little differently there. Should we abstract the namespace out to define it at the start depending on your provider (along with the cluster name), then just use the variable here?

Reply:
On OpenShift every component occupies a dedicated namespace. I think adding one variable for this would make sense for sure, but docs are a poor abstraction, so variable can be fragile, and they also make things a bit hard to read or even error-prone for the user. However, for the lack of a better abstraction, adding a shell variable is certainly plausible 👍

Reply:
OK, I didn't realize there may be an Openshift-specific angle to this, I thought it might just be a GKE holdover. Appreciate the explanation.

I also don't fully understand the scheduling piece I'm referring to here on how k8s treats namespaces other than `kube-system`, or for that matter whether there's any OpenShift-specific logic like that. Does OpenShift deploy OVN in a dedicated OVN namespace too? I guess if it's good enough for the default CNI then it should likely be good enough for Cilium.

Reply:
> I also don't fully understand the scheduling piece

I think `kube-system` is no longer meant to be special: https://github.com/kubernetes/kubernetes/pull/60653

> Does OpenShift deploy OVN in a dedicated OVN namespace too? 

Yes, and e.g. Multus also gets a dedicated namespace.

Reply:
Oh wow, kube-system has not meant to be special since 1.11! Goes to show how rumours can persist unless they're stopped short, thanks for finding & sharing that.